### PR TITLE
Remove xml feeds

### PR DESCRIPTION
**Pull request by mrducky4**
Address issue #1224
Remove the old XML feeds, keeping the RSS feeds.
Remove the feed type parameter from backend.php.
Replace the old feed icon with the fontawesome RSS icon.
Add backend.php to pageload_smoketest.py.

_Update from cpeel_: Sandbox avail at https://www.pgdp.org/~cpeel/c.branch/remove_xml_feeds

### CODE REVIEW

**Review by chrismiceli — APPROVED** [review]

File: feeds/backend.php
Comment (cpeel) [inline feeds/backend.php:31]: `$charset` is always going to be `UTF-8` since the conversion to Unicode in 2020. I have a branch ready to remove this global from the codebase but was waiting on this PR to land first so I didn't generate conflicts. Lets drop the variable here and just put `UTF-8` in the XML header as a literal.

File: feeds/backend.php
Comment (cpeel) [inline feeds/backend.php:40]: Please add type definitions here:
```suggestion
function generate_rss_feed(string $content, string $site_name, string $code_url, string $site_manager_email_addr): string
```

File: pinc/showavailablebooks.inc
Comment (cpeel) [inline pinc/showavailablebooks.inc:125]: We can delete `graphics/rss.gif` in this PR too. Need to leave the XML one around for a bit though since the Teams and Members code uses it.

**Review by cpeel — COMMENTED** [review]
This is great and I had just a few suggestions -- thank you for working on it!

File: feeds/backend.php
Comment (mrducky4) [reply feeds/backend.php]: Done. I changed it in the HTTP response header too.

File: pinc/showavailablebooks.inc
Comment (mrducky4) [reply pinc/showavailablebooks.inc:125]: Done.

File: feeds/backend.php
Comment (mrducky4) [reply feeds/backend.php]: Done. Is this a new practice? I didn't see it in [DP Code Best Practices](https://www.pgdp.net/wiki/DP_Code_Best_Practices).

File: styles/themes/charcoal.css
Comment (srjfoo) [inline styles/themes/charcoal.css:1795]: After looking at this actual size on Charcoal, I like it less than I initially thought I would (it looks fine on the white-based themes) -- I've done a bit of research, and it looks like there is no one true color for the RSS icon as far as the internet is concerned, so, if I may suggest an alternative for Charcoal ... I tried #ff9a00 (another, somewhat lighter version), and then tried mixing the two. #fd9205 looks good to me, and is slightly darker but not as vibrant against the very dark grey background as the existing color.

IIRC, it means adding an override to `charcoal.less`. I don't feel like it has to happen in this PR, but wanted to mention it here first.

File: pinc/showstartexts.inc
Comment (mrducky4) [reply pinc/showstartexts.inc:41]: Good question. I followed [fontawesome documentation](https://fontawesome.com/v5/icons/rss-square?f=classic&s=solid), without digging into the underlying mechanism. I also saw other examples in the code that already use `<i>` for fontawesome. Just now I also found documentation that says you can use `<span>` instead of `<i>`, and maybe other elements too, but fontawesome recommends using one element type consistently.

File: feeds/backend.php
Comment (mrducky4) [reply feeds/backend.php:14]: I checked on the specification for the Expires header and found GMT is required. [RFC9111](https://www.rfc-editor.org/rfc/rfc9111#field.expires) refers to [RFC9110](https://www.rfc-editor.org/rfc/rfc9110#section-5.6.7), which says it must be GMT. I assume the same is true for Last-Modified, but I didn't check that one.

File: styles/themes/charcoal.css
Comment (mrducky4) [reply styles/themes/charcoal.css:1795]: I tried it with #fd9205, I have no preference of one color over the other. It is an easy change, so I'm happy to add this color for the charcoal theme in this PR if that's what the team prefers.

File: pinc/showstartexts.inc
Comment (srjfoo) [reply pinc/showstartexts.inc:41]: 😁 -- shows you how alert I am, or I'd have seen it in earlier fa-related code. If I'd noticed then, I'd probably have suggested `<span>` just out of persnicketyness (since it doesn't imply a particular treatment of the enclosed character), but I agree. Whatever we use, we should be consistent.

File: styles/themes/charcoal.css
Comment (srjfoo) [reply styles/themes/charcoal.css:1795]: Screenshots for everyone to look at:

Current sandbox (#f26522), charcoal: 
image

Charcoal with #fd9205: 
image

default: 
image

default with #fd9205: 
image

(Wow ... I had no idea that the background for charcoal was so close to the background (in dark mode) for github 😁 )

File: feeds/backend.php
Comment (cpeel) [reply feeds/backend.php]: It is a new practice. I'll get the best practices code updated - thanks for calling it out.

File: pinc/showstartexts.inc
Comment (cpeel) [reply pinc/showstartexts.inc:41]: Yes, let's stick with `<i>` as that's what we use most other places.

File: styles/themes/charcoal.css
Comment (cpeel) [reply styles/themes/charcoal.css:1795]: I think the current color on white looks good and defer to @srjfoo on what color she thinks is best for charcoal. If we can get it into this PR that would be appreciated.

File: feeds/backend.php
Comment (cpeel) [reply feeds/backend.php]: Added this (brief) section on typing here: https://www.pgdp.net/wiki/DP_Code_Best_Practices#Typing

File: styles/themes/charcoal.css
Comment (mrducky4) [reply styles/themes/charcoal.css]: I updated the PR to use #fd9205 for charcoal.

**Review by cpeel — APPROVED** [review]

**Review by srjfoo — APPROVED** [review]

**Pull request merged**
